### PR TITLE
fix(content-conformance): adjust markdown reporter

### PR DESCRIPTION
**Pull request by BRKalow**
🎟️ [Asana Task]()

---

## Description

Fix formatting of the markdown reporter. Shifts parse errors to render above the table as they have the tendency to be multiple lines.

Example output: https://github.com/hashicorp/mktg-content-workflows/actions/runs/4118101205#summary-11178176279

## PR Checklist 🚀

- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Write a useful description (above) to give reviewers appropriate context.
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/content-conformance/src/reporters/markdown-reporter.ts [hunk 1]
```diff
@@ -43,8 +51,17 @@ export function markdownReporter(files: VFile[]) {
...
+      tableRows.push(`| ${columns.join(' | ')} |`)
     })
+
+    // Global parse errors
+    if (parseErrors.length > 0) {
+      lines.push(...parseErrors, '')
+    }
+
+    // Table
+    lines.push('| Position | Description | Rule |', '|---|---|---|')
```
Comment (thiskevinwang) [inline packages/content-conformance/src/reporters/markdown-reporter.ts:63]:
```suggestion
    lines.push('| Position | Description | Rule |', '| --- | --- | --- |')
```

🤔 (needs snapshot update)

Comment (thiskevinwang) [reply packages/content-conformance/src/reporters/markdown-reporter.ts:63]: SIde: Wonder if this GitHub actions `Summary` class is at all useful:
https://github.com/actions/toolkit/blob/main/packages/core/src/summary.ts#L229-L255

Comment (BRKalow) [reply packages/content-conformance/src/reporters/markdown-reporter.ts:63]: @thiskevinwang It seems to be working without the spaces here 🤔

And yeah I saw that after I had initially made this! I'm not sure I want to add @actions/core as a dependency here, but I'm tempted.

**Review by thiskevinwang — APPROVED** [review]
LGTM! Just left inconclusive observations

**Pull request merged**